flink-streaming-connectors/flink-connector-elasticsearch2/src/main/java/org/apache/flink/streaming/connectors/elasticsearch2/ElasticsearchSink.java

```diff
@@ -177,7 +180,72 @@ public void open(Configuration configuration) {
 		if (LOG.isInfoEnabled()) {
 			LOG.info("Created Elasticsearch TransportClient {}", client);
 		}
+	}
+
+	@Override
+	public void invoke(final T element) {
+		ParameterTool params = ParameterTool.fromMap(userConfig);
+
+		if (params.has(CONFIG_NO_OF_CONN_RETRIES) && params.getInt(CONFIG_NO_OF_CONN_RETRIES) > 0) {
```

Comment:
This is two hashmap lookups for each element. The check can be done only once in the open method